### PR TITLE
Refactor markdown-engine.ts by extracting custom markdown features into separate files

### PR DESCRIPTION
**Pull request by kachkaev**
In response to https://github.com/shd101wyy/mume/issues/33

### CODE REVIEW

**Comment by kachkaev** [comment]
I moved custom markdown features such as code fencing into separate files, which reduced the size of `markdown-engine.ts` from 2944 to 2599 lines of code. There're still quite a few options to make `markdown-engine.ts` easier to edit, however, this initial refactoring can already give an advantage because it makes new markdown features easier to implement.

File: src/custom-markdown-features/critic-markup.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ * {==[text]==}          	<mark>[text]</mark>	                  \hl{[text]}
+ * {>>[text]<<}          	<aside>[text]</aside>                	\marginpar{[text]}
+*/
+import { MarkdownIt } from 'markdown-it'
+import { MarkdownEngineConfig } from '../markdown-engine-config'
+import parseMath from '../parse-math'
+
+export default (md: MarkdownIt, config: MarkdownEngineConfig) => {
+  // @ts-ignore
+  md.inline.ruler.before('strikethrough', 'critic-markup', (state, silent) => {
```
Comment (kachkaev) [inline src/custom-markdown-features/critic-markup.ts]: @shd101wyy could you please explain why this function returns boolean? According to a [type definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f1f971c41a7a98820b8439ef206b98a380982079/types/markdown-it/index.d.ts#L99-L101) in [@types/markdown-it](https://www.npmjs.com/package/@types/markdown-it) it is expected to return void. I had to add `// @ts-ignore` in several feature files to ignore build errors meanwhile. Is there any place in mume or markdown-it that is using the returned true / false? I could not find it.

**Comment by kachkaev** [comment]
I've been using prettier [Prettier](https://prettier.io/) in several javascript projects and quite enjoyed the way it enforces order in all the code. It turned out that prettier began to support typescript a few months back, so I added it to this PR. Two new commits add Prettier together to [tslint](https://github.com/palantir/tslint), which is a helpful tool to spot imperfections that prettier cannot fix automatically. With these two new dependencies, an editor (Atom in my case) becomes much more responsive as the new code gets written. The more errors and warnings "as you go", the fewer errors in runtime and the easier it is to collaborate due to a community-enforced code standard.

**Comment by kachkaev** [comment]
Great to hear that @shd101wyy! Will you be happy if I simply press "merge pull request" or do you expect me to rebase etc.?

**Comment by shd101wyy** [comment]
@kachkaev I think simply "merge pull request" is good enough. Thanks.

**Comment by kachkaev** [comment]
Done! Looking forward to send more PRs after the new release! Just holding on for now to avoid too much complexity in the git tree :–)

**Pull request merged**